org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/SemanticHighlightingReconciler.java

```diff
@@ -537,13 +537,23 @@ public void install(JavaEditor editor, ISourceViewer sourceViewer, SemanticHighl
 		fSourceViewer= sourceViewer;
 
 		if (fEditor instanceof CompilationUnitEditor) {
-			((CompilationUnitEditor)fEditor).addReconcileListener(this);
-		} else if (fEditor == null) {
```

Comment:
This was I believe always wrong since if `scheduleJob()` is run when `fEditor` is `NULL`, it will thrown NPE there.